### ISSUE TITLE
Issues in the medianCV section

### ISSUE THREAD

**Issue by OrhunKok** [UCLouvain-CBIO/SCP.replication#1]
As is, the Replicate_specht2019v3.Rmd does not run. The problem occurs in the section where you compute medianCV. First, the computeMedianCV_SCoPE2 function is built to compute the median CV per cell computed using at least 5 observations, but I believe that according to the SCOPE2 paper, it should be 15 observations. This section also causes all the assays to be removed:
```r
scp <- scp[, scp$MedianCV < 0.365 & 
               scp$SampleType %in% c("Macrophage", "Monocyte"), ]
```
The plot that shows protein quantification for single-cells needs to use .MedianCV not MedianCV in:
```r
ggplot(aes(x = MedianCV, 
               fill = SampleType))
```
To solve the issue with computeMedianCV_SCoPE2 I used the standardised function in SCP, the medianCVperCell function, like this: 
```r
scp <- medianCVperCell(scp,
                       i = "peptides",
                       groupBy = "protein",
                       nobs = 15, 
                       norm = "div.median",
                       na.rm = TRUE,
                       colDataName = "MedianCV")
scp <- scp[, !is.na(scp$MedianCV) & scp$MedianCV > 0.365, ]
```

This also allowed me to have an nobs parameter of 15 in the function compared to the computeMedianCV_SCoPE2 function.

**Comment by lgatto** [comment]
Indeed, I hit the same issue yesterday. We are on it. Thanks for the report!

**Comment by cvanderaa** [comment]
Thanks @OrhunKok for reporting this. Here are my answer to your questions:

> First, the computeMedianCV_SCoPE2 function is built to compute the median CV per cell computed using at least 5 observations, but I believe that according to the SCOPE2 paper, it should be 15 observations.

I think you are referring to the last filtering step described in [Table 2](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02267-5/tables/2) of the paper, where they keep proteins that have been quantified in at least 15 cells. This step is actually performed by `filterNA` later on in the vignette. This is not linked to computing CVs. 

The authors computed the CV of protein only if at least 5 observations, that is 5 peptides, are available for that protein. You can see this [here in their code](https://github.com/SlavovLab/SCoPE2/blob/2d12001bc3bf393c56a22723c217d7af206a2b42/code/SCoPE2_analysis.R#L341-L343). I think this is a reasonable number, as you won't have an accurate estimate of the CV with fewer peptides and you might remove too many proteins when increasing the required number of peptides. 


>  This section also causes all the assays to be removed [...] The plot that shows protein quantification for single-cells needs to use .MedianCV not MedianCV

I think this is the same issue that @lgatto ran into. This is a version issue because I recently decided to rename `.MedianCV` to `MedianCV`. I'm very sorry about this. I now realize the latest version available on Bioconductor devel is `1.1.2`. To make the script run you should have at least `1.1.3`. I made the necessary push to Bioconductor, it should be available from there soon. Meanwhile, you can download the latest version (`1.1.4`) from this repo using:

```r
BiocManager::install("UCLouvain-CBIO/scp")
```
To make sure you have the correct version loaded, check your `sessionInfo()` and make sure you see `scp_1.1.4`. Let me know if this helps.
